card/mod/admin/spec/set/self/admin_spec.rb

```diff
@@ -9,45 +9,62 @@
   end
 
   context "#update" do
-    before do
-      @admin = Card[:admin]
-    end
+    let(:admin) {Card[:admin]}
 
-    it "triggers empty trash (with right params)" do
+    def run_admin_task task
       Card::Auth.as_bot do
-        Card["A"].delete!
+        Card::Env.params[:task] = task
+        admin.update_attributes({})
+      end
+    end
+
+    it "clearing trash is denied" do
+      expect { run_admin_task :empty_trash }
+        .to raise_error Card::Error::Oops, /The admin task 'empty trash'/
+    end
+
+    it "clearing history is denied" do
+      expect { run_admin_task :clear_history }
+        .to raise_error Card::Error::Oops, /The admin task 'clear history'/
+    end
+
+    context "irreversible tasks allowed" do
+      around(:example) do |example|
+        Cardio.config.allow_irreversible_admin_tasks = true
+        example.run
+        Cardio.config.allow_irreversible_admin_tasks = false
+      end
+
+      it "triggers empty trash (with right params)" do
+        Card::Auth.as_bot {Card["A"].delete!}
+
         expect(Card.where(trash: true)).not_to be_empty
-        Card::Env.params[:task] = :empty_trash
-        @admin.update_attributes({})
+        run_admin_task :empty_trash
         expect(Card.where(trash: true)).to be_empty
       end
-    end
 
-    # NOTE: I removed this functionality for now, because I don't think we
-    # should have web access to admin functions that can incur actual data loss.
```

Comment:
You only removed the method but not the link. I brought the method back but disabled it by default with an error message that explains how to turn it on.